src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -62,7 +62,7 @@ protected virtual string GetResourcePrefix(TypeInfo typeInfo)
                 throw new ArgumentNullException(nameof(typeInfo));
             }
 
-            return GetResourcePrefix(typeInfo, _hostingEnvironment.ApplicationName, _resourcesRelativePath);
+            return GetResourcePrefix(typeInfo, new AssemblyName(typeInfo.Assembly.FullName).Name, _resourcesRelativePath);
```

Comment:
It is possible and correct for GetResourcePrefix to return typeInfo.FullName. In the case that we don't have a Resource path it would return something like `Microsoft.Extensions.Localization.Tests.ResourceManagerStringLocalizerFactoryTest` which is how it will be stored in the assembly. Or are you asking something else?